### PR TITLE
Ivy-blue-email-add-administrative-details

### PR DESCRIPTION
**Pull request by 20chen-7**
# Description
Screenshot 2024-03-31 at 1 25 20 PM

## Main changes explained:
- Add administrative content and previous blue square historyInfringements
…

## How to test:
1. check into current branch
2. Gmail API
Before testing, you need to set up your own test email id to send the mail notification.
This is only for test purposes. Once tested you can revert these changes..

To send an email using the Gmail API with OAuth2 and nodemailer, follow the video below and set it up.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

Fill in the these variable values from the above setup in the .env file of the backend.
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = true
3. comment this to run every minute
Screenshot 2024-03-31 at 1 49 23 PM
add test user email here:
Screenshot 2024-03-31 at 1 29 58 PM
change the email address you wanna receive the blue square email
Screenshot 2024-03-31 at 1 39 06 PM
for the frontend, based on @Abdel Lallouache's recommendation(https://highest-good.slack.com/files/U051KB9CPNW/F06S268Q2H0/expl.mp4), we should comment `if (!validateDateIsNotBeforeStartOfCurrentWeek(requestData)) return;`
Screenshot 2024-03-31 at 1 44 18 PM
add dashed line
Screenshot 2024-03-31 at 3 18 03 PM

5. do `npm install` and `...` to run this PR locally
6. Clear site data/cache
7. log as admin user 
8. go to dashboard→ user management→ assign the test user day off(→…
Screenshot 2024-03-31 at 1 49 35 PM
The date is one week before the current week(explanation in the video provided by Abdel Lallouache) 
Thank @Abdel Lallouache's instructions and @DiegoSalas27 's help in modifying eslint file to get rid of some formatting issues brought by the previous version(which I do not want to change a lot).

### CODE REVIEW

**Review by ImzIssa — COMMENTED** [review]
Hey, I've been trying to review your PR but I'm not successful. I've managed to get and set up the API Keys but I can't get the email.

If you could provide more detailed steps on how to test this PR that would be great.

File: .eslintrc [hunk 1]
```diff
@@ -17,7 +17,14 @@
     "template-curly-spacing": "off",
     "indent": "off",
     "linebreak-style": 0,
-    "no-console": "off"
+    "no-console": "off",
```
Comment (ptrpengdev) [inline .eslintrc:20]: I noticed that you have made modifications to the ESLint rules. Perhaps we should make these changes in a separate PR and leave this PR to the add-admin-detail feature only.

File: src/helpers/userHelper.js [hunk 2]
```diff
@@ -94,11 +94,13 @@ const userHelper = function () {
     timeRemaining,
     coreTeamExtraHour,
     requestForTimeOffEmailBody,
+    administrativeContent,
```
[the remark this answers is not in the document]
Comment (ptrpengdev) [reply src/helpers/userHelper.js:106]: Reviewed on 4/19. The issue persists. getInfringementEmailBody is invoked in the timeEntryController with four parameters.

image

**Review by metaphor987 — COMMENTED** [review]
Hi Ivy,

I have set up the gmail api following the instructions, and can successfully receive the blue square emails every minute. These emails are sent from and to the test user and are BCC'd to the email I set up in the "emailSender" field, and I am unsure if this behavior is correct. The email content includes the expected administrative details.
Screenshot 2024-04-20 at 8 49 01 PM

During further testing, I encountered issues when attempting to open the video link for the "Add new time off request" feature(https://highest-good.slack.com/files/U051KB9CPNW/F06S268Q2H0/expl.mp4). 
Screenshot 2024-04-20 at 8 47 11 PM

Could you please provide the link again or provide additional clarification on how to set the "Date of leave" and what behavior should be expected afterward? In my testing, I set the date of leave to be today and also a week before today, but I have not observed any changes and continue to receive emails every minute.

Thank you!

**Comment by 20chen-7** [comment]
> Hi Ivy,
> 
> I have set up the gmail api following the instructions, and can successfully receive the blue square emails every minute. These emails are sent from and to the test user and are BCC'd to the email I set up in the "emailSender" field, and I am unsure if this behavior is correct. The email content includes the expected administrative details. Screenshot 2024-04-20 at 8 49 01 PM
> 
> During further testing, I encountered issues when attempting to open the video link for the "Add new time off request" feature(https://highest-good.slack.com/files/U051KB9CPNW/F06S268Q2H0/expl.mp4). Screenshot 2024-04-20 at 8 47 11 PM
> 
> Could you please provide the link again or provide additional clarification on how to set the "Date of leave" and what behavior should be expected afterward? In my testing, I set the date of leave to be today and also a week before today, but I have not observed any changes and continue to receive emails every minute.
> 
> Thank you!

Thank you for your review! and I found the latest frontend has already fixed this bug, So we do not need to modify the frontend anymore. As for the bcc issue, this is defined as the initial dev version, for this part, we only change the onecommunityglobal email to our email, just for test.

**Review by abdel-lall — CHANGES_REQUESTED** [review]
Hi @20chen-7, According to the description, users should not receive administrative details. However, during testing, both the admin email and user email received administrative details.
Owner Email:
Screenshot 2024-04-20 234158

User Email:
Screenshot 2024-04-20 234150

**Comment by 20chen-7** [comment]
> Hi @20chen-7, According to the description, users should not receive administrative details. However, during testing, both the admin email and user email received administrative details. Owner Email: Screenshot 2024-04-20 234158
> 
> User Email: Screenshot 2024-04-20 234150

This is out of the scope of this PR.

**Review by ptrpengdev — COMMENTED** [review]
Hi @20chen-7 

I noticed that you updated the getInfringementEmailBody function within the userHelper. Please make sure that the getInfringementEmailBody invoked in the timeEntryController (line 308) is not affected by the updated params.

**Comment by 20chen-7** [comment]
> Hi @20chen-7
> 
> I noticed that you updated the getInfringementEmailBody function within the userHelper. Please make sure that the getInfringementEmailBody invoked in the timeEntryController (line 308) is not affected by the updated params.

So based on PR673((https://github.com/OneCommunityGlobal/HGNRest/pull/673/files)), adding new input parameter(requestForTimeOffEmailBody) will not change the blue square mail, the same logic for adding the administrative content.

**Comment by one-community** [comment]
Thank you all, merging!

**Review by one-community — APPROVED** [review]

**Pull request merged**